tracing/src/main/java/com/palantir/tracing/DetachedSpan.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tracing;
+
+import com.palantir.tracing.api.SpanType;
+
+/**
+ * Span operation which is not bound to thread state, and can measure operations which
+ * themselves aren't bound to individual threads.
+ */
+public interface DetachedSpan {
```

Comment:
We may want an API to create a child `DetachedSpan` from this one without modifying state.
This could be accomplished indirectly using the following, however it requires an intermediate operation. I am hesitant to add an API to apply our span to a thread without also applying a new span.
```java
try (SpanToken ignored detached.startSpanOnCurrentThread("operation") {
    return Tracer.startDetachedSpan("nextOperation");
}
```